streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -640,15 +636,16 @@ private void prepareTopic(final Map<String, InternalTopicMetadata> topicPartitio
                 throw new org.apache.kafka.streams.errors.TopologyBuilderException(String.format("%sTopic [%s] number of partitions not defined", logPrefix, topic.name()));
             }
 
-            topicsToMakeReady.put(topic, numPartitions);
+            topic.setNumberOfPartitions(numPartitions);
+            topicsToMakeReady.put(topic.name(), topic);
             topicNamesToMakeReady.add(topic.name());
         }
 
         if (!topicsToMakeReady.isEmpty()) {
             internalTopicManager.makeReady(topicsToMakeReady);
 
             // wait until each one of the topic metadata has been propagated to at least one broker
-            while (!allTopicsCreated(topicNamesToMakeReady, topicsToMakeReady)) {
+            while (!allTopicsCreated(topicNamesToMakeReady, topicsToMakeReady.values())) {
```

Comment:
This sounds weird to me: why not just using `topicsToMakeReady` map than keeping `topicsToMakeReady` and `topicNamesToMakeReady`?